components/brave_shields/browser/site_specific_script_service.cc

```diff
@@ -0,0 +1,142 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_shields/browser/site_specific_script_service.h"
+
+#include <algorithm>
+#include <utility>
+
+#include "base/base_paths.h"
+#include "base/bind.h"
+#include "base/files/file_util.h"
+#include "base/json/json_reader.h"
+#include "base/logging.h"
+#include "base/macros.h"
+#include "base/memory/ptr_util.h"
+#include "base/strings/utf_string_conversions.h"
+#include "base/threading/thread_restrictions.h"
+#include "brave/browser/brave_browser_process_impl.h"
+#include "brave/components/brave_shields/browser/ad_block_service.h"
+#include "brave/components/brave_shields/browser/local_data_files_service.h"
+#include "brave/components/brave_shields/browser/dat_file_util.h"
+
+namespace brave_shields {
+
+SiteSpecificScriptService::SiteSpecificScriptService()
+    : weak_factory_(this) {
+  DETACH_FROM_SEQUENCE(sequence_checker_);
+}
+
+SiteSpecificScriptService::~SiteSpecificScriptService() {
+}
+
+SiteSpecificScriptService::SiteSpecificScriptRule::SiteSpecificScriptRule() =
```

Comment:
The destructor can probably be a default, but we absolutely need the SiteSpecificScriptRule ones because it's a complex type and Chromium hard-prevents you from compiling them without an explicit constructor.

Reply:
OK, gotcha. I thought it might be that.